pkg/ratelimiter/ratelimiter.go

```diff
@@ -0,0 +1,13 @@
+package ratelimiter
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// RateLimiter is responsible for allowing records and in and keep track of the
+// records already allowed. It denies records when the limit is reached.
+type RateLimiter interface {
```

Comment:
This interface seems to be a proxy one to me. How do we use the polymorphism it provides?